src/genie/libs/parser/iosxr/show_dampening.py

```diff
@@ -0,0 +1,241 @@
+            out = output
+            
+        result_dict = {}
+        result_dict.setdefault('interface', {})
+        
+        # GigabitEthernet0/0/0/1                                               2389 YES
+        # POS0/2/0/0                  <base>             ppp                      0 NO 
+        p1 = re.compile(r'^(?P<interface>.{1,26}) +(?P<prot>.{1,20}) +(?P<cap>.{1,10}) +(?P<pen>\d+) +(?P<sup>YES|NO)')
+
+        for line in out.splitlines():
+            if line:
+                line = line.strip()
+            else:
+                continue
+            
+            m = p1.match(line)
+            
+            if m:
+                group = m.groupdict()
+                interface=group['interface'].rstrip()
+                
+                # set "not_present" on any empty fields
+                for empty in group.keys():
+                    if group.get(empty).startswith(' '):
+                        group[empty]="not_present"
+                protocol=group['prot'].rstrip()
+                
+                if result_dict['interface'].get(interface)==None:
+                    result_dict['interface'][interface]={}
+                    result_dict['interface'][interface]['protocol']={}
+                result_dict['interface'][interface]['protocol'][protocol]={}
+                result_dict['interface'][interface]['protocol'][protocol].update({'capsulation': group['cap']})  
+                result_dict['interface'][interface]['protocol'][protocol].update({'penalty': int(group['pen'])})
+                result_dict['interface'][interface]['protocol'][protocol].update({'suppressed': group['sup'].lower()})                                     
+                continue
+        return result_dict
+
+class ShowImDampeningIntfSchema(MetaParser):
+    schema = {
+        'interface': {
+            Any(): {
+            'dampening_status': str,    
+            Optional('currently_suppressed'): str,
+            Optional('half_life'): int,
+            Optional('max_supress_time'): int,
+            Optional('penalty'): int,
+            Optional('reuse'): int,
+            Optional('suppress'): int,
+            Optional('suppressed_secs_remaining'): int,
+            Optional('underlying_state'): str,
+            Optional('protocol'): 
+                     {Any():
+                        {Optional('protocol_capsulation'): str, 
+                        Optional('protocol_penalty'): int,
+                        Optional('protocol_suppression'): str,
+                        Optional('protocol_suppression_timer'): int,
+                        Optional('protocol_underlying_state'): str,
+                      },
+                   },
+                }
+        }
+    }
+    
+class ShowImDampeningIntf(ShowImDampeningIntfSchema):
+    """ Parser for show im dampening interface {interface} """
+    
+    cli_command = ['show im dampening interface {interface}']
+    
+    def cli(self, interface=None, output=None):
```

Comment:
lets make interface mandatory. Below we can then remove the `if interface:` line

Reply:
Certainly, no problem have done that now in the updated version.